### PR TITLE
[v6r15] Added dry run for CS agents

### PR DESCRIPTION
**Pull request by fstagni**
WIP because still un-tested, but you're welcome for commenting already

### CODE REVIEW

File: ConfigurationSystem/ConfigTemplate.cfg [hunk 1]
```diff
@@ -40,9 +40,11 @@ Agents
     AutoModifyUsers = False
     AutoSuspendUsers = False
     VO = None
+    DryRun = False
```
Comment (andresailer) [inline ConfigurationSystem/ConfigTemplate.cfg]: I would set the CS default to True, so when the agent is installed it is not doing anything until it is explicitly enabled.

File: ConfigurationSystem/Agent/VOMS2CSAgent.py [hunk 2]
```diff
@@ -18,11 +16,29 @@
...
-  def initialize( self ):
+  def __init__( self, *args, **kwargs ):
+    """ Defines default parameters
+    """
 
     self.__voDict = {}
+    self.__adminMsgs = {}
+    self.csapi = None
+    self.voChanged = False
+    self.dryRun = True
```
Comment (andresailer) [inline ConfigurationSystem/Agent/VOMS2CSAgent.py]: I would set this to `False`, or people updating will not notice that their agents are now behaving differently, and existing agents don't have the option set in the CS.

Comment (fstagni) [inline ConfigurationSystem/Agent/VOMS2CSAgent.py]: So, basically you would swap the flags I have put here. I would be more in favor of having a default that is "same" everywhere, and at the same time make very clear points in the release notes for such a change.

**Comment by fstagni** [comment]
Not anymore a WIP. Please review. For now, I have put a "DryRun=True" by default (if it stays, it needs advertisement).

**Comment by andresailer** [comment]
I wanted to say that the documentation for the agents should be updated too, but there isn't any for most of them. https://github.com/DIRACGrid/DIRAC/tree/rel-v6r15/docs/source/AdministratorGuide/Configuration/ConfReference/Systems/Configuration/Agents

File: ConfigurationSystem/ConfigTemplate.cfg [hunk 3]
```diff
@@ -40,9 +40,11 @@ Agents
     AutoModifyUsers = False
     AutoSuspendUsers = False
     VO = None
+    DryRun = True
```
Comment (andresailer) [inline ConfigurationSystem/ConfigTemplate.cfg]: There should also be "DryRun" for the Bdii2CSAgent

**Comment by fstagni** [comment]
Thanks, done. I also modified the documentation.

File: docs/source/AdministratorGuide/Configuration/ConfReference/Systems/Configuration/Agents/Bdii2CSAgent/index.rst [hunk 4]
```diff
@@ -1,5 +1,5 @@
-Systems / Configuration / <INSTANCE> / Agents /CE2CSAgent - Sub-subsection
-==========================================================================
+Systems / Configuration / <INSTANCE> / Agents /Bdii2CSAgent - Sub-subsection
+====================================================++======================
 
 CE2CSAgent is the agent in charge of update sites parameters configuration for a specific VO.
```
Comment (andresailer) [inline docs/source/AdministratorGuide/Configuration/ConfReference/Systems/Configuration/Agents/Bdii2CSAgent/index.rst]: Replace `CE2CSAgent` here

File: docs/source/AdministratorGuide/Configuration/ConfReference/Systems/Configuration/Agents/Bdii2CSAgent/index.rst [hunk 5]
```diff
@@ -1,5 +1,5 @@
-Systems / Configuration / <INSTANCE> / Agents /CE2CSAgent - Sub-subsection
-==========================================================================
+Systems / Configuration / <INSTANCE> / Agents /Bdii2CSAgent - Sub-subsection
+====================================================++======================
```
Comment (andresailer) [inline docs/source/AdministratorGuide/Configuration/ConfReference/Systems/Configuration/Agents/Bdii2CSAgent/index.rst]: `++` signs for some reason

**Comment by fstagni** [comment]
Review OK now?

**Comment by zmathe** [comment]
review ok

**Pull request merged**